Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1418,10 +1419,7 @@ namespace AzToolsFramework
                     parentEntitiesToUpdate.insert(GetEntityById(parentEntityId));
                 }
 
-                commonOwningInstance->get().DetachEntity(entityId).release();
```

Comment:
What was the difference between `release` and `reset`?

Reply:
In this case, DetachEntity() returns a unique ptr which holds the ownership of the memory. `release` just releases the ownership but does not actually free the memory.

In the previous workflow, we then depend on ComponentApplication::DeleteEntity call to delete the entity object and free the memory. In the new changes, we just call reset by ourselves on the unique ptr. `reset` frees the memory.

Reply:
I think common use case is assign to a local variable after `release`, otherwise the memory is leaked (no longer being tracked).